docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
+1. As a user, I should be able to select the default file format for my report, contextual to my selected source (PDF, PNG, CSV, Excel, etc)
+1. As a user, I should be able to add a `Header` and `Footer` to PDF and PNG reports.
+    1. As a user, I should see Report title and a sample paragraph in the `header` by Default.
+    1. As a user, I should see date, time and report name in the `footer` by Default.
+1. As a user, I should be able to select which charts/visualizations I can include in a PDF or PNG report. (p2)
+
+##### Report trigger
+1. As a user, I should be able to select a report trigger type that belongs to one of two categories: `Schedule`, and `Alert`.
+1. As a user, when I select `Schedule`, I should be able to select a request time of now, future date, recurring, or custom cron expression (similar to alerting).
+    1. As a user, if I select `now`, I should be able to generate a report immediately after saving the report definition.
+    1. As a user, if I select `future date`, I should be able to configure the date and time the report will be generated.
+    1. As a user, if I select `recurring`, I should be able to configure a recurring schedule.
+    1. As a user, if I select `custom cron`, I should be able to configure a cron-based schedule.
+    1. As a user, if I select `recurring` or `custom cron`, I should be able to configure an end date.
+1. As a user, when I select `Alert`, I should be able to select the alert a report gets triggered from.
+
+##### Delivery settings
+1. As a user, I should be able to deliver a report to Kibana and/or Email recepients.
+1. As a user, when I set Kibana as a delivery channel, I should be able to select which Kibana users the report gets delivered to.
+1. As a user, when I set email as a delivery channel, I should be able to set recepients, email subject, and email body.
+1. As a user, I should be able to insert a file URL refecence to the email body.
+1. As a user, I should be able to insert the report source URL reference to the email body.
+1. As a user, I should be able to attach the report file to an email.
+1. As a user, I should be able to set the file format of report attachment (pdf, png | csv, excel, depending on the source).
+
+#### Report creation
+1. As a user, when I create a report definition, the definition settings are saved and show up as a list item in the ‘Report definitions’ list.
+1. As a user, when a report definition has been triggered via schedule or alert, I should be able to see an instance of the report in my "Reports" list, and any users I send the report to, will also see a report instance in their repective "Reports" list page.
+1. As a user, when a report is created from a triggered definition condition, I should receive a notification on all configured delivery channels (Email, chime, slack, etc)
+1. As a user, when a report is created from a triggered definition condition, I should receive a kibana toast alert letting me know that the report is available.
```

Comment:
Toast alert will pop up no matter where we are in Kibana?

Reply:
We might need to think about this one a bit more. It would be annoying to be inundated by toasts. Maybe only pop a toast when a user has been added as a recepient via the "share" option in the report details page. (So automated and scheduled reports don't create extra noise)

Reply:
We will change this user to something along this lines:

1. As a user, when a new report is available and I'm outside the reporting plugin, there will be some kind of indicator on the reporting plugin icon to let the user know that there are new reports available. We shouldn't generate toasts.

Reply:
So to clarify- if I'm on Kibana but not on the reporting plugin, the only notification I might see is one that's on the Reporting plugin icon in the sidebar?